### PR TITLE
*: deprecate sysvar tidb_enable_tiflash_pipeline_model

### LINKED ISSUE #47138: remove sys var tidb_enable_tiflash_pipeline_model
## Enhancement
After discussion, decide to remove `tidb_enable_tiflash_pipeline_model`. Why:
1. Normally user doesn't care which tiflash execution engine they are using.
2. TiFlash Pipeline execution engine is only to make it possible to implement tiflash resource control.
3. Too many switches will increase the mental burden of users.
So we decide to remove `tidb_enable_tiflash_pipeline_model`

### PR DESCRIPTION
**Pull request by guo-shaoge**
### What problem does this PR solve?

Issue Number: close #47138

Problem Summary: As issue said. Deprecate `tidb_enable_tiflash_pipeline_model` sys var.

### What is changed and how it works?
revert #44467 and #46709

```
  new tidb: sysvar is deprecated, it means sysvar will not dispatch to tiflash
  new tiflash: will ignore tidb dispatched var, check [tiflash code](https://github.com/pingcap/tiflash/pull/8116/files#diff-10f2455b9bff49016c20c69e9f2ba9c30325b530a37e232ea3aac3f2b879db88L161)

  new tidb -dispatch req-> new tiflash: ok.
  old tidb -dispatch req-> new tiflash: tidb dispatch sysvar to tiflash to enable or disable pipeline model, but tiflash will ignore.
  new tidb -dispatch req-> old tiflash: tidb will not dispatch sysvar to tiflash, tiflash will not use pipeline mode
  old tidb -dispatch req-> old tiflash: ok
```
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by XuHuaiyu — APPROVED** [review]

File: executor/set_test.go [hunk 1]
```diff
@@ -2157,3 +2157,10 @@ func TestSetMppExchangeCompressionModeVariable(t *testing.T) {
 		require.Equal(t, warnings[0].Err.Error(), "mpp exchange compression won't work under current mpp version 0")
 	}
 }
+
+func TestDeprecateEnableTiFlashPipelineModel(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec(`set @@global.tiflash_enable_pipeline_model = 1`)
```
Comment (SeaRise) [inline executor/set_test.go:2164]:
```suggestion
	tk.MustExec(`set @@global.tidb_enable_tiflash_pipeline_model = 1`)
```

**Review by SeaRise — APPROVED** [review]
Others LGTM

**Pull request merged**